multinet/api/views/common.py

```diff
@@ -56,7 +81,7 @@ def get_queryset(self):
 
         parent_query_dict = self.get_parents_query_dict()
         workspace = get_object_or_404(
-            Workspace.objects.select_related('owner'), name=parent_query_dict['workspace__name']
+            Workspace.objects.select_related('owner'), name=parent_query_dict[self.workspace_field]
```

Comment:
Does this limit it to only owned workspaces? We have a notion of maintainers, would we care about letting maintainers modify the sessions, too?

Reply:
> Does this limit it to only owned workspaces?

No, what `select_related()` does is ask the query engine to prefetch the named, "related" column (by doing a JOIN operation, etc.) so that downstream if we reference the workspace object's `owner` field, a separate database query won't be performed.

> We have a notion of maintainers, would we care about letting maintainers modify the sessions, too?

Take a look at this block of code from later in this function:

https://github.com/multinet-app/multinet-api/blob/200c7e8fbe6f6fc9c0c2fb504f4aadf80413b8da/multinet/api/views/common.py#L95-L101

`workspace_role` is a permission level descriptor which (as you can see in the first part of the code) depends on the workspace and the user making this request. If that is anything besides `None` (i.e., READER or higher), _or_ if the requesting user owns the workspace (there's that reference to `.owner` I mentioned above), then this function grants access to the query results.

Note that this is all part of `WorkspaceChildMixin`, which implements workspace-based filtering for routes that have the `/workspaces/{workspace_id}/` prefix. For the ability to write to workspaces when you have maintainer or higher access, we use a decorator like the one used here:

https://github.com/multinet-app/multinet-api/blob/200c7e8fbe6f6fc9c0c2fb504f4aadf80413b8da/multinet/api/views/session.py#L45-L65